cirq-core/cirq/protocols/resolve_parameters.py

```diff
@@ -177,6 +177,11 @@ def resolve_parameters(
     if isinstance(val, (list, tuple)):
         return cast(T, type(val)(resolve_parameters(e, param_resolver, recursive) for e in val))
 
+    getter = getattr(val, '_is_parameterized_', None)
+    result = NotImplemented if getter is None else getter()
+    if result is False:
+        return val
```

Comment:
nit: I'd suggest using a different var name for the function attribute in this case. Also, I'd suggest not setting `result` since that is used for the actual result of parameter resolution below, so using it for the return value of this method here feels odd
```suggestion
    is_parameterized = getattr(val, '_is_parameterized_', None)
    if is_parameterized is not None and not is_parameterized():
        return val
```

Reply:
done. I'm comparing `is_parameterized() is False` instead to distinguish with the case when `is_parameterized()` can return `NotImplemented`.